cmd/dist-receiver/main.go

```diff
@@ -0,0 +1,325 @@
+	}
+
+	stageNow = 1
+
+	<-sig
+
+	if err := connectToPeers(ctx, node, targets); err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to connect to peers: %s\n", err)
+	}
+
+	cid := icorepath.New(*cidArg)
+
+	if err := node.Dht().Provide(ctx, cid); err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to seed the resource file: %s\n", err)
+	}
+
+	if err := node.Pin().Add(ctx, cid); err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to pin the resource file: %s\n", err)
+	}
+
+	rootNode, err := node.Unixfs().Get(ctx, cid)
+
+	if err != nil {
+		panic(fmt.Errorf("could not get file with CID: %s", err))
+	}
+
+	if err := os.RemoveAll(*fileName); err != nil {
+		panic(fmt.Errorf("could not clean previous temporary file: %s", err))
+	}
+
+	err = files.WriteTo(rootNode, *fileName)
+	if err != nil {
+		panic(fmt.Errorf("could not write out the fetched CID: %s", err))
+	}
+
+	err = os.MkdirAll(*targetDir, os.ModePerm)
+	if err != nil {
+		panic(fmt.Errorf("failed to create target directory: %s", err))
+	}
+
+	if err := extractTarGz(*fileName, *targetDir); err != nil {
+		panic(fmt.Errorf("failed to uncompress resource file: %s", err))
+	}
+
+	stageNow = 2
+
+	<-sig
+}
+
+func createNode(ctx context.Context, repoPath string) (*core.IpfsNode, error) {
+	// Open the repo
+	repo, err := fsrepo.Open(repoPath)
+	if err != nil {
+		return nil, err
+	}
+
+	// Construct the node
+	nodeOptions := &core.BuildCfg{
+		Online:  true,
+		Routing: libp2p.DHTOption, // This option sets the node to be a full DHT node (both fetching and storing DHT Records)
+		// Routing: libp2p.DHTClientOption, // This option sets the node to be a client DHT node (only fetching records)
+		Repo: repo,
+	}
+
+	return core.NewNode(ctx, nodeOptions)
+}
+
+func spawnEphemeral(ctx context.Context, bootstrap string) (icore.CoreAPI, *core.IpfsNode, error) {
+	if err := setupPlugins(""); err != nil {
+		return nil, nil, err
+	}
+
+	repoPath, err := createTempRepo(bootstrap)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create temp repo: %s", err)
+	}
+
+	node, err := createNode(ctx, repoPath)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	api, err := coreapi.NewCoreAPI(node)
+
+	return api, node, err
+}
+
+func setupPlugins(externalPluginsPath string) error {
+	plugins, err := loader.NewPluginLoader(filepath.Join(externalPluginsPath, "plugins"))
+	if err != nil {
+		return fmt.Errorf("error loading plugins: %s", err)
+	}
+
+	if err := plugins.Initialize(); err != nil {
+		return fmt.Errorf("error initializing plugins: %s", err)
+	}
+
+	if err := plugins.Inject(); err != nil {
+		return fmt.Errorf("error initializing plugins: %s", err)
+	}
+
+	return nil
+}
+
+func createTempRepo(bootstrap string) (string, error) {
+	repoPath, err := os.MkdirTemp("", "ipfs-shell")
+	if err != nil {
+		return "", fmt.Errorf("failed to get temp dir: %s", err)
+	}
+
+	cfg, err := config.Init(io.Discard, 2048)
+	if err != nil {
+		return "", err
+	}
+
+	var bs []string
```

Comment:
I wonder if the bs variable is necessary here

Reply:
Replaced with `cfg.Bootstrap = nil`. The purpose is to remove all preset bootstrap nodes since we don't mean to share the images across the internet.